[Dev→master] Bounding Box and App Bar

cre8ivepark opened the pull request:
![mrtk_appbar_boundingbox_examplescene](https://user-images.githubusercontent.com/13754172/38342211-d5fc9bfc-3831-11e8-803d-a48abf73138e.jpg)

Overview
---
Migration to master. Bounding Box & App Bar with an example scene.

Requires
---
Example scene uses TwoHandManipulatable script. #1849 should be merged first.
Included bug fix for AppBar highlighting #1895 

Test scene
---
Assets/HoloToolkit-Examples/UX/Scenes/BoundingBoxGizmoExample.unity

StephenHodgson commented:
Blocked by #1849

davidkline-ms commented:
Adding this to the May 2018 release.

StephenHodgson commented:
@davidkline-ms if we move this to the May 2018 release, will it won't compile without #1849 also getting merged into this branch

davidkline-ms commented:
@StephenHodgson,
I periodically update may18_dev with Patch4_Dev changes. Just did that, in fact with #1917

davidkline-ms commented on Assets/HoloToolkit/UX/Scripts/BoundingBoxes/BoundingBoxGizmoHandle.cs:
```diff
@@ -0,0 +1,398 @@
+
+namespace HoloToolkit.UX.BoundingBoxes
+{
+    public class BoundingBoxGizmoHandle : MonoBehaviour, IInputHandler, ISourceStateHandler
```
Please add documentation (summary blocks, etc) to this file.
Thx!

davidkline-ms commented on Assets/HoloToolkit/UX/Scripts/BoundingBoxes/BoundingBoxGizmoHandle.cs:
```diff
@@ -0,0 +1,398 @@
+{
+    public class BoundingBoxGizmoHandle : MonoBehaviour, IInputHandler, ISourceStateHandler
+    {
+        public enum RotationType
```
If these enums are intended to be public, please move them outside of the class.

davidkline-ms commented on Assets/HoloToolkit/UX/Scripts/BoundingBoxes/Duplicator.cs:
```diff
@@ -1,13 +1,11 @@
-namespace HoloToolkit.Unity.UX
+namespace HoloToolkit.UX.BoundingBoxes
 {
     public class Duplicator : MonoBehaviour
```
Please add documentation (summary blocks, etc) to this file.
Thx!

davidkline-ms commented:
Moving back to 2017.2.1.4. This change is fairly small and 100% useful

davidkline-ms commented on Assets/HoloToolkit/UX/Scripts/AppBar/AppBarButton.cs:
```diff
@@ -0,0 +1,184 @@
+using HoloToolkit.Unity.Buttons;
+using UnityEngine;
+
+namespace HoloToolkit.Unity.UX
```
Please add summary and other comments to facilitate documentation generation.

Thanks!

davidkline-ms commented on Assets/HoloToolkit/UX/Scripts/BoundingBoxes/BoundingBox.cs:
```diff
@@ -245,14 +147,71 @@ protected set
+        #region protected
+
+        [Header("Objects")]
+        [Tooltip("The target object")]
```
Please add doc comments (summary blocks, etc)

davidkline-ms commented on Assets/HoloToolkit/UX/Scripts/BoundingBoxes/BoundingBoxGizmoHandle.cs:
```diff
@@ -0,0 +1,398 @@
+
+namespace HoloToolkit.Unity.UX
+{
+    public class BoundingBoxGizmoHandle : MonoBehaviour, IInputHandler, ISourceStateHandler
```
Please add doc comments

cre8ivepark commented:
- Added summary blocks to AppBarButton, AppBar, BoundingBox, BoundingBoxGizmoHandle.
- Created separate enum files for the BoundingBoxGizmoHandle.
- Reverted BoundingBox and Duplicator

cre8ivepark commented:
Added/updated README files. Updated landing page links,

- [BoundingBox and AppBar](https://github.com/cre8ivepark/MixedRealityToolkit-Unity/blob/Patch4_Dev_BoundingBoxAppBar/Assets/HoloToolkit-Examples/UX/Readme/README_BoundingBoxGizmoExample.md)
- [TwoHandManipulation](https://github.com/cre8ivepark/MixedRealityToolkit-Unity/blob/Patch4_Dev_BoundingBoxAppBar/Assets/HoloToolkit-Examples/Input/Readme/README_TwoHandManipulationTest.md)
-[Landing page examples links update 
](https://github.com/cre8ivepark/MixedRealityToolkit-Unity/tree/Patch4_Dev_BoundingBoxAppBar)

davidkline-ms commented on Assets/HoloToolkit-Examples/UX/Readme/README_InteractableObjectExample.md:
```diff
@@ -1,54 +1,54 @@
+This example scene demonstrates how to make objects interactable using [CompoundButton](/Assets/MixedRealityToolkit/UX/Scripts/Buttons/CompoundButton.cs) script scries. With CompoundButton script, you can make any object interactable with differentiated visual state for the HoloLens' gesture input or immersive headset's motion controller input state. For more information please see ['Interactable ...
 
-The scene also demonstrates how to use [Receiver](/Assets/HoloToolkit/UX/Scripts/Receivers/InteractionReceiver.cs) to handle events comging from multiple objects, in a single script. This is especially useful when you have multiple buttons or interactable objects in a scene. 
+The scene also demonstrates how to use [Receiver](/Assets/MixedRealityToolkit/UX/Scripts/Receivers/InteractionReceiver.cs) to handle events comging from multiple objects, in a single script. This is especially useful when you have multiple buttons or interactable objects in a scene. 
```
Master branch is still using HoloToolkit[-Examples]

davidkline-ms commented on Assets/HoloToolkit-Examples/UX/Readme/README_ObjectCollection.md:
```diff
@@ -6,34 +6,34 @@ Object collection is a script which helps you lay out an array of objects in pre
 
 ## Ways to use Object collection ##
-You can find the examples in the scene **ObjectCollection_Examples.unity**. In this scene, you can find the **ObjectCollection.cs** script under **Assets/HoloToolkit/UX/Scripts/Collections**
+You can find the examples in the scene **ObjectCollection_Examples.unity**. In this scene, you can find the **ObjectCollection.cs** script under **Assets/MixedRealityToolkit/UX/Scripts/Collections**
```
Master branch is still using HoloToolkit[-Examples]
